File: sdks/python/src/pricewatcha/job_status.py
"""Job status helpers shared by SDK polling logic."""

from __future__ import annotations

from typing import Any

TERMINAL_STATUSES = frozenset({"completed", "failed"})
ACTIVE_STATUSES = frozenset({"queued", "running", "processing"})


def is_terminal_job_status(status: Any) -> bool:
    return isinstance(status, str) and status in TERMINAL_STATUSES


def is_active_job_status(status: Any) -> bool:
    return isinstance(status, str) and status in ACTIVE_STATUSES

File: sdks/typescript/src/job-status.ts
import type { JobStatus } from "./types.js";

/** Job reached a terminal scrape outcome. */
export function isTerminalJobStatus(status: JobStatus): boolean {
  return status === "completed" || status === "failed";
}

/** Job is still in the ingestion pipeline (includes status `processing`). */
export function isActiveJobStatus(status: JobStatus): boolean {
  return status === "queued" || status === "running" || status === "processing";
}
